src/app_selection.rs

```diff
@@ -63,6 +62,7 @@ pub async fn try_get_valid_app(
                 // If we got a response from Holochain, then we have a chance of finding the app.
                 // Update the app info cache and search again.
 
+                let mut installed_apps = installed_apps.write().await;
                 *installed_apps = new_installed_apps.clone();
                 choose_unique_app(&dna_hash, &coordinator_identifier, &installed_apps)?.clone()
```

Comment:
This write lock could be downgraded here into a read lock. I don't think that it makes a difference right now as this is the only place that it's used, and a write lock is always required first, but maybe in the future.

```suggestion
                choose_unique_app(
                    &dna_hash,
                    &coordinator_identifier,
                    &installed_apps.downgrade(),
                )?
                .clone()
```

Reply:
That's a good point. I was thinking the lock was not going to be held much longer because the choose is quick, but it's an easy fix for a little less time holding the lock